### PR TITLE
Refactoring + int64 support

### PR DESCRIPTION
**Pull request by PhilippMDoerner**
Did some refactoring to the gType macro and added int64 support which I needed myself for one of my types.

### CODE REVIEW

File: src/typehelpers.nim [hunk 1]
```diff
@@ -48,29 +50,36 @@ proc gType*(ty: NimNode): tuple[kind: TyKind, ty: NimNode] =
...
-    #   if ty.getTypeImpl[0].getType.strval == "string":
-    #     return (TyString, ty.getTypeImpl())
-    return (TyRefObj, ty.getTypeImpl[0].getImpl[2][2]) # [2])
-
-  else: return (TyUnsupported, newNimNode(nnkNone))
+
+  let typeImpl = ty.getTypeImpl()
+  case typeImpl.kind:
+    of nnkSym:
+      let typeTyKind: TyKind = parseEnum[TyKind](typeImpl.strVal())
```
Comment (enthus1ast) [inline src/typehelpers.nim:49]: clever!

**Review by enthus1ast — CHANGES_REQUESTED** [review]
The tests are failing, could you have a look again :-)  

```nim
C:\Users\david\projects\nisane1\src>nim c -r nisane.nim
Hint: used config file 'C:\Users\david\.choosenim\toolchains\nim-1.6.0\config\nim.cfg' [Conf]
Hint: used config file 'C:\Users\david\.choosenim\toolchains\nim-1.6.0\config\config.nims' [Conf]
...........................................................................
C:\Users\david\projects\nisane1\src\nisane.nim(23, 46) Warning: `typed` will change its meaning in futur
e versions of Nim. `void` or no return type declaration at all has the same meaning as the current meani
ng of `typed` as return type declaration. [Deprecated]
C:\Users\david\projects\nisane1\src\nisane.nim(179, 21) Warning: type pragmas follow the type name; this
 form of writing pragmas is deprecated [Deprecated]
..............................
C:\Users\david\.choosenim\toolchains\nim-1.6.0\lib\impure\db_sqlite.nim(184, 1) Warning: Circular depend
ency detected. `codeReordering` pragma may not be able to reorder some nodes properly [User]
...
ObjectTy
  Empty
  Empty
  RecList
    IdentDefs
      Sym "first"
      Sym "string"
      Empty
    IdentDefs
      Sym "second"
      Sym "string"
      Empty
    IdentDefs
      Sym "third"
      Sym "float"
      Empty
    IdentDefs
      Sym "forth"
      Sym "int"
      Empty
stack trace: (most recent call last)
C:\Users\david\projects\nisane1\src\nisane.nim(121, 29) ct
C:\Users\david\projects\nisane1\src\nisane.nim(208, 10) template/generic instantiation of `ct` from here

C:\Users\david\projects\nisane1\src\nisane.nim(121, 29) Error: cannot get child of node kind: nnkEmpty



C:\Users\david\projects\nisane1\src>nim c -r typehelpers.nim
Hint: used config file 'C:\Users\david\.choosenim\toolchains\nim-1.6.0\config\nim.cfg' [Conf]
Hint: used config file 'C:\Users\david\.choosenim\toolchains\nim-1.6.0\config\config.nims' [Conf]
.........................................................................
stack trace: (most recent call last)
C:\Users\david\projects\nisane1\src\typehelpers.nim(91, 21) foo1
C:\Users\david\projects\nisane1\src\typehelpers.nim(73, 25) gType
C:\Users\david\projects\nisane1\src\typehelpers.nim(62, 49) gType
C:\Users\david\.choosenim\toolchains\nim-1.6.0\lib\std\enumutils.nim(66, 21) parseEnum
C:\Users\david\projects\nisane1\src\typehelpers.nim(106, 10) template/generic instantiation of `assert` from here
C:\Users\david\.choosenim\toolchains\nim-1.6.0\lib\std\enumutils.nim(66, 21) Error: unhandled exception: Invalid enum value: int [ValueError]

**Comment by PhilippMDoerner** [comment]
I'll take a look over lunch, sadly in meetings until then. I just had a realization where that issue might be coming from though!

**Comment by enthus1ast** [comment]
no need to hurry, i'm also at work :)

**Comment by PhilippMDoerner** [comment]
> The tests are failing, could you have a look again :-)

Done

**Comment by enthus1ast** [comment]
Thanks! Much cleaner now!

**Pull request merged**